Source/SourceKittenFramework/File.swift

```diff
@@ -16,36 +17,46 @@ import SWXMLHash
 // This file could easily be split up
 
 /// Represents a source file.
-public final class File {
+public final class File { // swiftlint:disable:this type_body_length
     /// File path. Nil if initialized directly with `File(contents:)`.
     public let path: String?
     /// File contents.
     public var contents: String {
         get {
-            if _contents == nil {
-                _contents = try! String(contentsOfFile: path!, encoding: .utf8)
+            _contentsQueue.sync {
+                if _contents == nil {
+                    _contents = try! String(contentsOfFile: path!, encoding: .utf8)
+                }
             }
             return _contents!
         }
         set {
-            _contents = newValue
+            _contentsQueue.sync {
+                _contents = newValue
```

Comment:
should this also invalidate `_lines`?

Reply:
You are right.
I'll make additional changes.

Reply:
Nice catch Sam 👍